commit 260ec04332714416be55f78b8314a0ca862c296c
Author: Jake <github@xvmc.com>
Date:   Thu Dec 19 01:39:09 2013 -0500

    Workaround for KnCMiner November batch
    
    There is a bug in FPGA firmware that cause I2C hang.
    - One possible workaround is to wait after every I2C access to Ericsson DCDC module. This works around partially works but still cause hang about once every 24 hours.
    - A more radical workaround is to just remove all I2C acces to Ericsson DCDC module. By doing this you cannot monitor Voltage from bfgminer but everything else works. So far this workaround completlly fix/hide the issue. No I2C hang for the last 6 days.

diff --git a/driver-knc.c b/driver-knc.c
index 55a4fea..381cf20 100644
--- a/driver-knc.c
+++ b/driver-knc.c
@@ -738,6 +738,7 @@ bool knc_get_stats(struct cgpu_info * const cgpu)
 		
 		if (0 == i % 0x30)
 		{
+			/*
 			if (ioctl(i2c, I2C_SLAVE, i2cslave_dcdc[die]))
 			{
 				applog(LOG_DEBUG, "%s: %s: Failed to select i2c slave 0x%x",
@@ -755,6 +756,10 @@ bool knc_get_stats(struct cgpu_info * const cgpu)
 			
 			volt    = (float)rawvolt * exp2(-10);
 			current = (float)knc_dcdc_decode_5_11(rawcurrent);
+			*/
+			
+			volt    = 1; // fake value
+			current = 1; // fake value
 			
 			applog(LOG_DEBUG, "%s: die %d %6.3fV %5.2fA",
 			       cgpu->dev_repr, die, volt, current);
